src/_pytest/mark/structures.py

```diff
@@ -386,6 +387,12 @@ def store_mark(obj, mark: Mark) -> None:
     This is used to implement the Mark declarations/decorators correctly.
     """
     assert isinstance(mark, Mark), mark
+
+    from ..fixtures import getfixturemarker
+
+    if getfixturemarker(obj) is not None:
+        warnings.warn(MARKED_FIXTURE, stacklevel=2)
```

Comment:
Are we generating this warning only once per fixture function?

Reply:
Should be one for each mark

Reply:
I'll try adding a test